src/argparse/argparse.lua

```diff
@@ -0,0 +1,35 @@
+local logger = require("logger/logger")
+
+local argparse = {}
+
+function argparse:parse(raw)
+  local args = {}
+
+  function parse_flag(flag)
+    for i=1,#raw do
```

Comment:
Oh you 1 based-indexing... 🥴

Reply:
Heh, I blame lua and love2d.  They give you a table that looks like this:

{
  1: "--debug",
  2: "--connect",
  3: "127.0.0.1",
  -1: "embedded boot.lua",
  -2: "C:\Program Files\LOVE\love.exe"
}

Reply:
Haha, no worries.